sql/src/main/java/io/crate/operation/collect/stats/StatsTables.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.collect.stats;
+
+import com.google.common.base.Supplier;
+import com.twitter.jsr166e.LongAdder;
+import io.crate.operation.reference.sys.job.JobContext;
+import io.crate.operation.reference.sys.job.JobContextLog;
+import io.crate.operation.reference.sys.operation.OperationContext;
+import io.crate.operation.reference.sys.operation.OperationContextLog;
+import org.elasticsearch.common.collect.Tuple;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * StatsTables is responsible for adding jobs and operations of that node.
+ * It also provides the functionality to expose that data for system tables,
+ * such as sys.jobs, sys.jobs_log, sys.operations and sys.operations_log;
+ */
+@ThreadSafe
+public class StatsTables {
+
+    private final Map<UUID, JobContext> jobsTable = new ConcurrentHashMap<>();
+    private final Map<Tuple<Integer, UUID>, OperationContext> operationsTable = new ConcurrentHashMap<>();
+
+    final AtomicReference<LogSink<JobContextLog>> jobsLog = new AtomicReference<>(NoopLogSink.instance());
+    final AtomicReference<LogSink<OperationContextLog>> operationsLog = new AtomicReference<>(NoopLogSink.instance());
+
+    private final JobsIterableGetter jobsIterableGetter;
+    private final OperationsIterableGetter operationsIterableGetter;
+    private final LongAdder activeRequests = new LongAdder();
+    private final Supplier<Boolean> enabled;
+
+    public StatsTables(Supplier<Boolean> enabled) {
```

Comment:
Another minor thing: This could be a BooleanSupplier (specific boolean primitive type variant). To make sure that it cannot return null.